test/06-pytest-args/CMakeLists.txt

```diff
@@ -0,0 +1,70 @@
+cmake_minimum_required(VERSION 3.20)
+
+project(TestPytestArgs)
+
+find_package(Pytest REQUIRED)
+
+enable_testing()
+
+# The --help flag will suppress test discovery
+pytest_discover_tests(
+	pytestargs_no_test_discovered_w_help
+	PYTEST_ARGS --help
+	DEPENDS test_basic.py
+)
+
+# The -v flag will suppress test discovery
+# as the output for the command: pytest --co -q
+# Is affected by this commandline flag
+pytest_discover_tests(
+	pytestargs_no_test_discovered_w_args_v
+	PYTEST_ARGS -v
+	DEPENDS test_basic.py
+)
+
+pytest_discover_tests(
+	pytestargs_no_test_discovered_w_args_v_s
+	PYTEST_ARGS -v -s
+	DEPENDS test_basic.py
+)
+
+# add_test(NAME TestPytestArgs.Validate.test_no_discovered_w_args_v_s
+#     COMMAND ${CMAKE_COMMAND}
+#     -D "TEST_PREFIX=\"\""
+#     -D "EXPECTED=\"\""
+#     -P ${CMAKE_CURRENT_SOURCE_DIR}/../01-modify-name/compare_discovered_tests.cmake
+# )
+
+pytest_discover_tests(
+	pytestargs_basic_args
+	PYTEST_ARGS -v -s -q
+	DEPENDS test_basic.py
+)
+
+add_test(NAME TestPytestArgs.Validate.test_basic
+    COMMAND ${CMAKE_COMMAND}
+    -D "TEST_PREFIX=pytestargs_basic_args.test_basic"
+    -D "EXPECTED=pytestargs_basic_args.test_basic"
+    -P ${CMAKE_CURRENT_SOURCE_DIR}/../01-modify-name/compare_discovered_tests.cmake
+)
+
+# Currently quoted args are not supported
+pytest_discover_tests(
+	pytestargs_basic_args_quoted
+	PYTEST_ARGS "-v -s -q"
+	DEPENDS test_basic.py
+)
+
+# semi-colon value to the PYTEST_ARGS is considered as a list
+pytest_discover_tests(
+	pytestargs_basic_args_semicolon
+	PYTEST_ARGS -v;-s;-q
+	DEPENDS test_basic.py
+)
+
+add_test(NAME TestPytestArgsSemicolon.Validate.test_basic
```

Comment:
These tests focus on arguments for discovery, but we could also design a test that fails unless a specific argument, such as `--capture=no`, is provided.

```python
import pytest

def test_requires_no_capture(request):
    """Fails unless '--capture=no' is passed to pytest."""
    if "--capture=no" not in request.config.invocation_params.args:
        pytest.fail("Test requires '--capture=no' to properly display output.")
    assert True
```

The `CMakeLists.txt` would be something like:

```cmake
cmake_minimum_required(VERSION 3.20)

project(TestExtraArgs)

find_package(Pytest REQUIRED)

enable_testing()

pytest_discover_tests(
    TestExtraArgs
    EXTRA_ARGS "--capture=no"
)
```